table/block_based/block_based_table_builder.cc

```diff
@@ -89,12 +92,12 @@ FilterBlockBuilder* CreateFilterBlockBuilder(
       partition_size = std::max(partition_size, static_cast<uint32_t>(1));
       return new PartitionedFilterBlockBuilder(
           mopt.prefix_extractor.get(), table_opt.whole_key_filtering,
-          filter_bits_builder, table_opt.index_block_restart_interval,
+          filter_bits_builder.release(), table_opt.index_block_restart_interval,
           use_delta_encoding_for_index_values, p_index_builder, partition_size);
     } else {
       return new FullFilterBlockBuilder(mopt.prefix_extractor.get(),
                                         table_opt.whole_key_filtering,
-                                        filter_bits_builder);
+                                        filter_bits_builder.release());
```

Comment:
Rather than the released pointer, maybe have a std::unique_ptr<>&& passed in and use std::move?  That is what is done in most places in the code.